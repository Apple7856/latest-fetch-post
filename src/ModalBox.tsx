import { Box, Dialog } from '@mui/material'
import React from 'react'

type jsonDataProps = {
    open: boolean
    jsonData: {
        title?: string
        author?: string
        url?: string
        created_at?: string
    }
    handleClose: ()=>void
}

const ModalBox = (props: jsonDataProps) => {
    return (
        <Dialog open={props.open} onClose={props.handleClose}>
            <Box data-testid="modalBox" sx={{ width: "300px", height: "300px", backgroundColor: "white"}}>
                {
                    JSON.stringify(props.jsonData)
                }
            </Box>
        </Dialog>
    )
}

export default ModalBox