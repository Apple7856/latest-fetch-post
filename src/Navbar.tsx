import { AppBar, Box, TextField, Toolbar, Typography } from '@mui/material'
import React from 'react'

type handleSearchProps = {
    handleSearch: (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>)=>void
}

const Navbar = (props:handleSearchProps) => {
    return (
        <Box>
            <AppBar>
                <Toolbar sx={{
                    display: "flex",
                    alignItems: "center",
                    "justifyContent": "space-around"
                }}>
                    <Typography variant="h6" data-testid="heading">Fetch-Post</Typography>
                    <TextField placeholder='Search...' data-testid="textField" onChange={(e)=>props.handleSearch(e)} type="text" sx={{ marginLeft: "10px", border: "2px solid white", borderRadius: "5px", backgroundColor: "white" }} />
                </Toolbar>
            </AppBar>
        </Box>
    )
}

export default Navbar