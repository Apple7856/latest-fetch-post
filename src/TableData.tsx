import { Box, FormControl, InputLabel, MenuItem, Select, SelectChangeEvent, Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@mui/material'
import React, { useState } from 'react'

type dataValue = {
    title?: string
    author?: string
    url?: string
    created_at?: string
}

type dataProps = {
    data: {
        title?: string
        author?: string
        url?: string
        created_at?: string
    }[]
    handleClick: (item: dataValue) => void
    searchValue: {
        title?: string
        author?: string
        url?: string
        created_at?: string
    }[]
    showData: boolean
}

const TableData = (props: dataProps) => {
    const [age, setAge] = useState("");
    function handleChange(e: SelectChangeEvent) {
        setAge(e.target.value as string);
    }
    return (
        <Box sx={{ marginTop: "70px" }}>
            <TableContainer>
                <Table>
                    <TableHead>
                        <TableRow>
                            <TableCell data-testid="title" sx={{ cursor:"pointer", color: "Red", fontWeight: "bold" }}>
                                Title
                            </TableCell>
                            <TableCell data-testid="author" sx={{ color: "Red", fontWeight: "bold" }}>Author</TableCell>
                            <TableCell data-testid="url" sx={{ color: "Red", fontWeight: "bold" }}>Url</TableCell>
                            <TableCell data-testid="created_at" sx={{ color: "Red", fontWeight: "bold" }}>Created_at</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {
                            props.showData ?
                                props.searchValue.map((item, index: number) => {
                                    return <TableRow key={index} onClick={() => props.handleClick(item)}>
                                        <TableCell>{item?.title}</TableCell>
                                        <TableCell>{item?.author}</TableCell>
                                        <TableCell>{item?.url}</TableCell>
                                        <TableCell>{item?.created_at}</TableCell>
                                    </TableRow>
                                })
                                :
                                props.data.map((item, index: number) => {
                                    return <TableRow key={index} onClick={() => props.handleClick(item)}>
                                        <TableCell>{item?.title}</TableCell>
                                        <TableCell>{item?.author}</TableCell>
                                        <TableCell>{item?.url}</TableCell>
                                        <TableCell>{item?.created_at}</TableCell>
                                    </TableRow>
                                })
                        }
                    </TableBody>
                </Table>
            </TableContainer>
        </Box>
    )
}

export default TableData;