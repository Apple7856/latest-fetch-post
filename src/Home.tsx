import { Box } from '@mui/material'
import axios from 'axios'
import React, { useEffect, useState } from 'react'
import ModalBox from './ModalBox'
import Navbar from './Navbar'
import TableData from './TableData'

type dataState = {
    title?: string
    author?: string
    url?: string
    created_at?: string
}

const Home = () => {
    const [data, setData] = useState<Array<dataState>>([]);
    const [jsonData, setJsonData] = useState<dataState>({});
    const [open, setOpen] = useState<boolean>(false);
    const [count, setCount] = useState<number>(0);
    const [searchValue, setSearchValue] = useState<Array<dataState>>([]);
    const [showData, setShowData] = useState<boolean>(false);
    function ApiCall() {
        axios({
            method: "GET",
            url: `https://hn.algolia.com/api/v1/search_by_date?tags=story&page=${count}`,
            responseType: "json"
        })
            .then((res) => {
                setData(data.concat(res?.data?.hits));
            })
            .catch((error) => {
                console.log(error);
            })
    }

    function timer() {
        setCount(prevCount => prevCount + 1);
    }

    function handleClick(item: dataState) {
        setJsonData(item);
        setOpen(true);
    }

    function handleScroll(e: any) {
        const bottom = Math.trunc(e.target.documentElement.scrollHeight - e.target.documentElement.scrollTop) === e.target.documentElement.clientHeight;
        if (bottom) {
            setCount(prevCount => prevCount + 1)
        }
    }

    function handleClose() {
        setOpen(false);
    }

    function handleSearch(e:any) {
        const inputWord = e.target.value;
        if (inputWord) {
            setShowData(true);
            const searchData = data.filter((item) => {
                if (item.title?.toLowerCase().includes(inputWord.toLowerCase()) || item.author?.toLowerCase().includes(inputWord.toLowerCase())) {
                    return item
                }
            })
            setSearchValue(searchData);
        } else {
            setShowData(false)
        }
    }

    useEffect(() => {
        ApiCall();
        const id = setInterval(timer, 10000);
        window.addEventListener("scroll", handleScroll, { passive: true });
        return () => {
            clearInterval(id);
            window.removeEventListener("scroll", handleScroll);
        }
    }, [count])

    return (
        <Box data-testid="homeBox">
            <Navbar handleSearch={handleSearch} />
            <TableData data={data} handleClick={handleClick} showData={showData} searchValue={searchValue} />
            <ModalBox open={open} jsonData={jsonData} handleClose={handleClose} />
        </Box>
    )
}

export default Home