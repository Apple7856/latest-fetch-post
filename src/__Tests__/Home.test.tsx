import { cleanup, render, act, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Home from '../Home';
import Navbar from '../Navbar';
import TableData from '../TableData';
import ModalBox from '../ModalBox';
import axios from 'axios';

beforeEach(() => {
    document.body.innerHTML = "";
})

afterEach(() => {
    cleanup();
})

describe("Home Components Test", () => {
    test("Box Test", () => {
        const component = render(<Home />)
        const box = component.getByTestId('homeBox');
        expect(box).toBeTruthy();
    })
    test("Create Snapshot Test", () => {
        act(() => {
            const component = render(<Home />)
            expect(component).toMatchSnapshot();
        })
    })
})

describe("Navbar Components Test", () => {
    test("Heading Test", () => {
        const component = render(<Navbar handleSearch={jest.fn()} />);
        const heading = component.getByTestId('heading');
        expect(heading.textContent).toBe('Fetch-Post');
    })
    test("Input Field Test", () => {
        const component = render(<Navbar handleSearch={jest.fn()} />);
        const textField = component.getByTestId('textField');
        expect(textField).toBeTruthy();
    })
    test("Create SnapShot Test", () => {
        act(() => {
            const component = render(<Navbar handleSearch={jest.fn()} />);
            expect(component).toMatchSnapshot();
        })
    })
})

describe("TableData Components Test", () => {
    const post = [
        {
            title: "abc",
            author: "xyz",
            url: "mvvc",
            created_at: "kmfk"
        },
        {
            title: "abc",
            author: "xyz",
            url: "mvvc",
            created_at: "kmfk"
        }
    ]
    test("Table Heading Test", () => {
        const component = render(<TableData data={post} handleClick={jest.fn()} showData={false} searchValue={post} />);
        const title = component.getByTestId('title');
        const author = component.getByTestId('author');
        const url = component.getByTestId('url');
        const created_at = component.getByTestId('created_at');
        expect(title.textContent).toBe("Title");
        expect(author.textContent).toBe("Author");
        expect(url.textContent).toBe("Url");
        expect(created_at.textContent).toBe("Created_at");
    })
    test("Create SnapShot Test", () => {
        act(() => {
            const component = render(<TableData data={post} handleClick={jest.fn()} showData={false} searchValue={post} />);
            expect(component).toMatchSnapshot();
        })
    })
})

describe("ModalBox Component Test", () => {
    const post = {
        title: "abc",
        author: "xyz",
        url: "mvvc",
        created_at: "kmfk"
    }

    test("Modal Data Test", () => {
        const component = render(<ModalBox open={true} jsonData={post} handleClose={jest.fn()} />);
        const modalBox = component.getByTestId('modalBox');
        expect(modalBox.textContent).toBe(JSON.stringify(post));
    })

    test("Create SnapShot Test", () => {
        act(() => {
            const component = render(<ModalBox open={true} jsonData={post} handleClose={jest.fn()} />);
            expect(component).toMatchSnapshot();
        })
    })
})

describe("Api Test", () => {
    const mockUrl = 'https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0';
    const mockData = { data: {} };
    const getData = jest.fn((url) => {
        return mockData;
    })
    test("Return Data", () => {
        expect(getData(mockUrl)).toBe(mockData);
    })
    test("Return url", () => {
        expect(getData).toBeCalledWith(mockUrl);
    })
    test("Api status", async () => {
        await act(async () => {
            const responce = await axios.get("https://hn.algolia.com/api/v1/search_by_date?tags=story&page=0");
            expect(responce.status).toBe(200);
        })
    })
})