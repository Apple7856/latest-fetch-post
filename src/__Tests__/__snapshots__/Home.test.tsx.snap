// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Components Test Create Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        data-testid="homeBox"
      >
        <div
          class="MuiBox-root css-0"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ucl8w4-MuiToolbar-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                data-testid="heading"
              >
                Fetch-Post
              </h6>
              <div
                class="MuiFormControl-root MuiTextField-root css-1kms70y-MuiFormControl-root-MuiTextField-root"
                data-testid="textField"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    placeholder="Search..."
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-nnbavb"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </header>
        </div>
        <div
          class="MuiBox-root css-fj59pj"
        >
          <div
            class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
          >
            <table
              class="MuiTable-root css-rqglhn-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                    data-testid="title"
                    scope="col"
                  >
                    Title
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                    data-testid="author"
                    scope="col"
                  >
                    Author
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                    data-testid="url"
                    scope="col"
                  >
                    Url
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                    data-testid="created_at"
                    scope="col"
                  >
                    Created_at
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              />
            </table>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      data-testid="homeBox"
    >
      <div
        class="MuiBox-root css-0"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ucl8w4-MuiToolbar-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              data-testid="heading"
            >
              Fetch-Post
            </h6>
            <div
              class="MuiFormControl-root MuiTextField-root css-1kms70y-MuiFormControl-root-MuiTextField-root"
              data-testid="textField"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-nnbavb"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </header>
      </div>
      <div
        class="MuiBox-root css-fj59pj"
      >
        <div
          class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="title"
                  scope="col"
                >
                  Title
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="author"
                  scope="col"
                >
                  Author
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="url"
                  scope="col"
                >
                  Url
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="created_at"
                  scope="col"
                >
                  Created_at
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            />
          </table>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ModalBox Component Test Create SnapShot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navbar Components Test Create SnapShot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ucl8w4-MuiToolbar-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              data-testid="heading"
            >
              Fetch-Post
            </h6>
            <div
              class="MuiFormControl-root MuiTextField-root css-1kms70y-MuiFormControl-root-MuiTextField-root"
              data-testid="textField"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-nnbavb"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-11b3ww9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-ucl8w4-MuiToolbar-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            data-testid="heading"
          >
            Fetch-Post
          </h6>
          <div
            class="MuiFormControl-root MuiTextField-root css-1kms70y-MuiFormControl-root-MuiTextField-root"
            data-testid="textField"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                placeholder="Search..."
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-nnbavb"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TableData Components Test Create SnapShot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-fj59pj"
      >
        <div
          class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        >
          <table
            class="MuiTable-root css-rqglhn-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="title"
                  scope="col"
                >
                  Title
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="author"
                  scope="col"
                >
                  Author
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="url"
                  scope="col"
                >
                  Url
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                  data-testid="created_at"
                  scope="col"
                >
                  Created_at
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  abc
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  xyz
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  mvvc
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  kmfk
                </td>
              </tr>
              <tr
                class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  abc
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  xyz
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  mvvc
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                >
                  kmfk
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-fj59pj"
    >
      <div
        class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          class="MuiTable-root css-rqglhn-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                data-testid="title"
                scope="col"
              >
                Title
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                data-testid="author"
                scope="col"
              >
                Author
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                data-testid="url"
                scope="col"
              >
                Url
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1h0dg0l-MuiTableCell-root"
                data-testid="created_at"
                scope="col"
              >
                Created_at
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                abc
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                xyz
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                mvvc
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                kmfk
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                abc
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                xyz
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                mvvc
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
              >
                kmfk
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
